app/src/main/java/com/kleshchin/danil/filemanager/ListAdapter.java

```diff
@@ -0,0 +1,105 @@
+package com.kleshchin.danil.filemanager;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+
+/**
+ * Created by Danil Kleshchin on 11.05.2017.
+ */
+
+public class ListAdapter extends BaseAdapter {
+
+    @NonNull
+    private ArrayList<File> file_;
+    @NonNull
+    private Context context_;
+
+    public ListAdapter(Context context, File file) {
+        context_ = context;
+        file_ = new ArrayList<>(Arrays.asList(file.listFiles()));
+        fillList(file_);
+    }
+
+    static class ViewHolder {
+        TextView fileName;
+        ImageView fileImage;
+    }
+
+    @Override
+    public int getCount() {
+        return file_.size();
+    }
+
+    @Override
+    public File getItem(int i) {
+        return file_.get(i);
+    }
+
+    @Override
+    public long getItemId(int i) {
+        return i;
+    }
+
+    @Override
+    public View getView(int i, View view, ViewGroup viewGroup) {
+        ViewHolder viewHolder;
+        if (view == null) {
+            view = LayoutInflater.from(context_).inflate(R.layout.list_items, viewGroup, false);
+            viewHolder = new ViewHolder();
+            viewHolder.fileName = (TextView) view.findViewById(R.id.fileName);
+            viewHolder.fileImage = (ImageView) view.findViewById(R.id.fileImage);
+            view.setTag(viewHolder);
+        } else {
+            viewHolder = (ViewHolder) view.getTag();
+        }
+        File file = getListItem(i);
+        Context context = viewGroup.getContext();
+        viewHolder.fileName.setText(file.getName());
+        if (file.isDirectory()) {
+            viewHolder.fileImage.setImageDrawable(context.getResources().getDrawable(R.drawable.folder));
+        } else {
+            viewHolder.fileImage.setImageDrawable(context.getResources().getDrawable(R.drawable.file));
+        }
+        return view;
+    }
+
+    File getListItem(int position) {
```

Comment:
It's not really necessary to have this wrapping method. getItem() has a strict meaning for adapters. You should use it directly because using getListItem makes me want to check how is it differ from getItem. And I don't like to dive into sources without a reason :smiling_imp: